build/tasks/test.js

```diff
@@ -1,38 +1,24 @@
-const gulp = require("gulp");
-const gutil = require("gulp-util");
-const Karma = require("karma").Server;
-const path = require("path");
 const ssvTools = require("@ssv/tools");
+const gulp = require("gulp");
+const jest = require('jest-cli');
+const runSeq = require("run-sequence");
 
-const config = require("../config");
-var args = require("../args");
+const args = require("../args");
 
-gulp.task("test", (cb) => {
-	runTests(true, cb);
-});
+const jestConfig = {
+	rootDir: 'source'
+};
 
-gulp.task("tdd", ["compile:test"], (cb) => {
-	runTests(false, cb);
+gulp.task("test", ["compile:test"], (cb) => {
+	return runSeq(
+		"compile:test",
+		"jest",
+		cb);
 });
 
-function runTests(singleRun, cb) {
-	gutil.log(gutil.colors.cyan(`Launching Karma... Reporters '${args.reporters}' ; Browsers '${args.browsers}'`));
-	new Karma({
-		configFile: path.join(__dirname, `../../${config.src.karmaConfig}`),
-		singleRun: singleRun,
-		autoWatch: !singleRun,
-		reporters: args.reporters,
-		browsers: args.browsers
-	}, (code) => {
-
-		// make sure failed karma tests cause gulp to exit non-zero
-		if (code === 1) {
-			gutil.log(gutil.colors.red("------- Error: unit test failed -------"));
-			return process.exit(1);
-		}
-		cb();
-	}).start();
-}
+gulp.task("jest", () => {
+	jest.runCLI({ config: jestConfig }, ".", () => done());
```

Comment:
you are invoking `done` but its not passed as a param

also can be written like
```js
gulp.task("jest", cb => jest.runCLI({ config: jestConfig }, ".", cb));
```

Reply:
done